src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -237,14 +237,22 @@ public SpawnResult exec(Spawn spawn, SpawnExecutionContext context)
               try (SilentCloseable c = Profiler.instance().profile("Remote.executeRemotely")) {
                 reply = remoteExecutor.executeRemotely(request);
               }
+
+              FileOutErr outErr = context.getFileOutErr();
+              String message = reply.getMessage();
+              if ((reply.getResult().getExitCode() != 0 ||
+                   reply.getStatus().getCode() != Code.OK.value()) && !message.isEmpty()) {
+                outErr.printErr(message + "\n");
```

Comment:
Just to clarify:
* For build failures Bazel will print this to the UI
* For test failures this will be the last line in the test.log

Correct?

Reply:
Exactly. It's printed in those two conditions.